### ISSUE TITLE
[BUG] ControllerX not working anymore with HA >2021.10 and Zigbee2MQTT 1.21.2 

### ISSUE THREAD

**Issue by rhannink** [xaviml/controllerx#394]
# Bug report

## Description

After upgrading HA to version 2021.10, my zigbee2MQTT integration with ControllerX is not working anymore

## Additional information

ControllerX 4.16.0
AppDaemon 4.1.0
HA > 2021.10
Z2M 1.21.2 
Mosquitto MQTT

Workaround is to switch from z2M to native MQTT

## AppDaemon app configuration

```yaml
Volume_keuken_app:
  module: controllerx
  class: E1744MediaPlayerController
  controller: sensor.volume1_action
  integration: z2m
  volume_steps: 50
  media_player: media_player.keuken_speaker
```

## Logs

```text

Appdaemon log
2021-10-12 10:07:26.916048 INFO AppDaemon: App initialization complete
2021-10-12 10:07:26.876061 INFO Volume_keuken_app: 🎮 ControllerX v4.16.0
2021-10-12 10:07:26.791160 INFO AppDaemon: Initializing app Volume_keuken_app using class E1744MediaPlayerController from module controllerx

Z2M log
Info MQTT publish: topic 'zigbee2mqtt/Volume1', payload '{"action":"","battery":74,"brightness":255,"linkquality":36,"rate":195,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Volume1/action', payload 'brightness_move_down'
Info MQTT publish: topic 'zigbee2mqtt/Volume1', payload '{"action":null,"battery":74,"brightness":235,"linkquality":36,"rate":195,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Volume1', payload '{"action":null,"battery":74,"brightness":215,"linkquality":36,"rate":195,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Volume1', payload '{"action":"brightness_stop","battery":74,"brightness":215,"linkquality":36,"rate":195,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Volume1', payload '{"action":"","battery":74,"brightness":215,"linkquality":36,"rate":195,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Volume1/action', payload 'brightness_stop'
```

**Comment by ristomatti** [comment]
Just noting that it works for me with Z2M 1.22-dev and HA 2021-11. I'm pretty sure I'm also using z2m integration since I couldn't get native MQTT working when I wanted to test it.

**Comment by rhannink** [comment]
@ristomatti,

I received following from the z2m maintainer (on HA forum):

_Hi @rhannink,

Interesting that you have more lag with MQTT than with HA. It should be the other way around. When set with MQTT, the controller communicates like the following:

controller > Zigbee2MQTT > MQTT > AppDaemon > ControllerX

However, the schema looks the following when setup with the sensor as you had:

controller > Zigbee2MQTT > MQTT > HA > AppDaemon > ControllerX

As you can see the HA adds an overhead, this is why I added the mqtt integration.

I will keep you updated once I investigte with my Symfonisk controller.

Regards,
Xavi M._

**Comment by xaviml** [comment]
Hi there,

Unfortunately, my E1744 controller does not work (I cannot pair it with Z2M). I will need to get another one to test this issue. I will keep you updated.

In the meantime, does the fix that I told @rhannink works for everyone?

Regards,
Xavi M.

**Comment by xaviml** [comment]
Hi all,

I was able to get my Symfonisk device working again. It all works as it used to be for me. These are my specs:
- Zigbee2MQTT version: 1.22.2
- Home Assistant: 2021.12.7
- ControllerX: 4.17.0
- AppDaemon add-on version: 0.7.2
- AppDaemon version: 4.1.0
- Symfonish firmware version: 2.1.022

I have tried using the `_action` sensor and the MQTT connection directly as well as trying it for a light and a media player. What I am currently doing now is updating the firmware to the latest version. I will keep you posted if it keeps working.

Regards,
Xavi M.
